### PR TITLE
Add support for linux/arm/v6

### PR DESCRIPTION
**Pull request by phdelodder**

### CODE REVIEW

**Comment by phdelodder** [comment]
[linux/arm/v6 9/9] RUN dpkg -i /deconz.deb &&     chown root:root /usr/bin/deCONZ* &&     setcap CAP_NET_BIND_SERVICE=+eip /usr/bin/deCONZ &&     rm -f /deconz.deb:
#44 0.323 dpkg: error processing archive /deconz.deb (--install):
#44 0.323  package architecture (armhf) does not match system (armel)
#44 0.384 Errors were encountered while processing:
#44 0.385  /deconz.deb

**Review by Jo-Be-Co — COMMENTED** [review]
Ok, with my suggestion I had initially only looked at how the grep functionality could be replaced.
1st) As the platform strings are well known the patterns could be streamlined somewhat more.
2nd) The urls used for v6 and v7 are the same - here could one pattern do the job.
Some examples:
```shell
case "{$PLATFROM}" in
  ...
  *v7*)
  *v6*)
    URL="http://deconz.dresden-elektronik.de/raspbian/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
    ;;
esac
```
```shell
case "{$PLATFROM}" in
  ...
  *v7*|*v6*)
    URL="http://deconz.dresden-elektronik.de/raspbian/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
    ;;
esac
```
```shell
case "{$PLATFROM}" in
  ...
  *v[67]*)
    URL="http://deconz.dresden-elektronik.de/raspbian/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
    ;;
esac
```
And with patterns matching on the end:
```shell
case "{$PLATFROM}" in
  */arm64)
    URL="http://deconz.dresden-elektronik.de/debian/${CHANNEL}/deconz_${DECONZ_VERSION}-debian-buster-${CHANNEL}_arm64.deb"
    ;;
  */amd64)
    URL="http://deconz.dresden-elektronik.de/ubuntu/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
    ;;
  */arm/v[67])
    URL="http://deconz.dresden-elektronik.de/raspbian/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
    ;;
esac
```
The first `*` could also be replaced by `linux` ... all in all, however, these are only refinements.

**Comment by phdelodder** [comment]
@Jo-Be-Co thanks I took you improved version and add it to the pull request. I'm still stuck with the arch issue.

**Comment by Jo-Be-Co** [comment]
> @Jo-Be-Co thanks I took you improved version and add it to the pull request.

Cheers. The main part was about not starting all these subprocesses by using grep.

> I'm still stuck with the arch issue.

On this I could not give any helpful input 😉

**Comment by manup** [comment]
The packages are all build for armhf (hardware floating point support), I'm no arm expert but if I understand the topic right this is not about v6/v7 but the OS should be armhf instead armel.

File: docker/download-deconz.sh [hunk 1]
```diff
@@ -6,18 +6,17 @@ DECONZ_VERSION=$1
...
-fi
-if echo "${PLATFORM}" | grep -qE "amd64";
-then
-  URL="http://deconz.dresden-elektronik.de/ubuntu/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
-fi
-if echo "${PLATFORM}" | grep -qE "v7";
-then
-  URL="http://deconz.dresden-elektronik.de/raspbian/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
-fi
+case "{$PLATFORM}" in
```
Comment (ablanchard) [inline docker/download-deconz.sh:9]: I think there is just a typo here. It should be
```suggestion
case "${PLATFORM}" in
```

File: docker/download-deconz.sh [hunk 2]
```diff
@@ -6,18 +6,17 @@ DECONZ_VERSION=$1
...
+    ;;
+  */amd64)
+    URL="http://deconz.dresden-elektronik.de/ubuntu/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
+    ;;
+  */arm/v[67])
+    URL="http://deconz.dresden-elektronik.de/raspbian/${CHANNEL}/deconz-${DECONZ_VERSION}-qt5.deb"
+    ;;
+esac
 
 curl -vv "${URL}" -o /deconz${DEV}.deb
```
Comment (Jo-Be-Co) [inline docker/download-deconz.sh:21]: Where does this `${DEV}` come from? The `Dockerfile` only uses `/deconz.deb` so `DEV` should never be set, should it?

Comment (phdelodder) [reply docker/download-deconz.sh:21]: It's a leftover from something I tested. I'm waiting on a specific armv6 build to update that script.

**Comment by senilio** [comment]
Soon to be 2 years old, I think it's safe to close this as not planned.

**Pull request closed without merging**